From bffe32cbe5d5cc6b3b315fcb7a2165c73e9a03ba Mon Sep 17 00:00:00 2001
From: John Morris <john@zultron.com>
Date: Fri, 31 Oct 2014 14:21:20 -0500
Subject: [PATCH] setup.py:  add 'test' command to setup.py

Signed-off-by: John Morris <john@zultron.com>
---
 setup.py | 29 +++++++++++++++++++++++++++--
 1 file changed, 27 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 131a3fb..6802d5e 100644
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,30 @@
 # Copyright (c) 2010 Jean-Paul Calderone
 # See LICENSE file for details.
 
-from distutils.core import Extension, setup
+from distutils.core import Extension, setup, Command
+import sys, distutils.util
+from unittest import TextTestRunner, TestLoader
+
+class TestCommand(Command):
+    user_options = [ ]
+
+    def initialize_options(self):
+        pass
+    def finalize_options(self):
+        sys.path.append(
+            'build/lib.%s-%s/signalfd' %
+            ( distutils.util.get_platform(), sys.version[0:3]))
+
+    def run(self):
+        '''Run unittests'''
+        import signalfd.test.test_signalfd
+        # testfiles = [ 'signalfd/test/test_signalfd.py' ]
+        # tests = TestLoader().loadTestsFromNames(testfiles)
+        # import signalfd.test
+        # tests = TestLoader().loadTestsFromModule(signalfd.test.test_signalfd)
+        # t = TextTestRunner(verbosity = 1)
+        # t.run(tests)
+
 
 setup(
     name="python-signalfd",
@@ -33,5 +56,7 @@ setup(
         "Programming Language :: Python :: 3.1",
         "Programming Language :: Python :: 3.2",
         "Topic :: Software Development :: Libraries :: Python Modules",
-        ])
+        ],
+    cmdclass = { 'test': TestCommand },
+    )
 
-- 
1.9.1

